tests/semantic_analyser.cpp

```diff
@@ -533,8 +533,9 @@ kprobe:f { @x = pid < 10000 ? "lo" : 2 }
 
 TEST(semantic_analyser, mismatched_call_types)
 {
+  // This assignment requires a cast, hint the requirement in the error message
   test_error("kprobe:f { @x = 1; @x = count(); }", R"(
-stdin:1:20-22: ERROR: Type mismatch for @x: trying to assign value of type 'count_t' when map already contains a value of type 'int64'
+stdin:1:20-22: ERROR: Type mismatch for @x: trying to assign value of type 'count_t' when map already contains a value of type 'int64', cast the 'count_t' value to 'int64'
 kprobe:f { @x = 1; @x = count(); }
```

Comment:
hmm, I'm not sure this is valid even with the cast. Doesn't the count operation need to be linked to a per_cpu map - meaning it doesn't really live as an isolated function. The map in this case is already a hash map.

Reply:
Very good point, we don't want to confuse our users with an incorrect hint!